### PR TITLE
fix pipeline and examples to reflect the clamp change in the Network

### PR DESCRIPTION
**Pull request by Hananel-Hazan**
1. fix pipeline to reflect the clamp change in the Network
2. fix examples

### CODE REVIEW

File: examples/cifar10/minimal_supervised_cifar10.py [hunk 1]
```diff
@@ -23,7 +23,10 @@
 for i in range(60000):
     # Choose an output neuron to clamp to spiking behavior.
     c = choice(10, size=1, replace=False)
-    clamp = {'Ae': 10 * labels[i].long() + Tensor(c).long()}
+    clamp = torch.zeros(pipeline.time, network.n_neurons, dtype=torch.uint8)
+    clamp[c, :] = 1
+    clamp_v = torch.zeros(pipeline.time, network.n_neurons, dtype=torch.float)
+    clamp_v[c, :] = 10 * labels[i].float() + Tensor(c).float()
```
Comment (djsaunde) [inline examples/cifar10/minimal_supervised_cifar10.py]: This should be `clamp_v[10 * labels[i].long() + Tensor(c).long(), :] = -52`

-52 is the threshold value.

File: examples/mnist/minimal_supervised_mnist.py [hunk 2]
```diff
@@ -21,8 +22,11 @@
 for i in range(60000):
     # Choose an output neuron to clamp to spiking behavior.
     c = choice(10, size=1, replace=False)
-    clamp = {'Ae': 10 * labels[i].long() + Tensor(c).long()}
-    
+    clamp = torch.zeros(pipeline.time, network.n_neurons,dtype=torch.uint8)
+    clamp[c, :] = 1
+    clamp_v = torch.zeros(pipeline.time, network.n_neurons,dtype=torch.float)
+    clamp_v[c,:] = 10 * labels[i].float() + Tensor(c).float()
```
Comment (djsaunde) [inline examples/mnist/minimal_supervised_mnist.py]: Same here.

File: examples/cifar10/minimal_supervised_cifar10.py [hunk 3]
```diff
@@ -23,7 +23,10 @@
 for i in range(60000):
     # Choose an output neuron to clamp to spiking behavior.
     c = choice(10, size=1, replace=False)
-    clamp = {'Ae': 10 * labels[i].long() + Tensor(c).long()}
+    clamp = torch.zeros(pipeline.time, network.n_neurons, dtype=torch.uint8)
+    clamp[c, :] = 1
```
Comment (djsaunde) [inline examples/cifar10/minimal_supervised_cifar10.py]: This should be `clamp[10 * labels[i].long() + Tensor(c).long(), :]`

File: examples/cifar10/minimal_supervised_cifar10.py [hunk 4]
```diff
@@ -23,7 +23,10 @@
 for i in range(60000):
     # Choose an output neuron to clamp to spiking behavior.
     c = choice(10, size=1, replace=False)
-    clamp = {'Ae': 10 * labels[i].long() + Tensor(c).long()}
+    clamp = torch.zeros(pipeline.time, network.n_neurons, dtype=torch.uint8)
```
Comment (djsaunde) [inline examples/cifar10/minimal_supervised_cifar10.py]: Could write `50` instead of `pipeline.time`...

File: bindsnet/pipeline/__init__.py [hunk 5]
```diff
@@ -124,9 +124,13 @@ def step(self, **kwargs) -> None:
 
         Keyword arguments:
 
-        :param Dict[str, torch.Tensor] clamp: Mapping from layer names to neurons to clamping to spiking.
+        :param Dict[str, torch.Tensor] clamp: Mapping of layer names to T/F if neuron at time t should be "clamp" to a
```
Comment (djsaunde) [inline bindsnet/pipeline/__init__.py]: This documentation needs a rewrite. I'll do it when I have time.

File: examples/cifar10/minimal_supervised_cifar10.py (hunk 1 above)
Comment (Hananel-Hazan) [reply examples/cifar10/minimal_supervised_cifar10.py]: 10 * labels[i].long() + Tensor(c).long() > then index of clamp_v

Comment (djsaunde) [reply examples/cifar10/minimal_supervised_cifar10.py]: What? `labels[i]` is in `range(10)`, and `c` is in `range(10)`, so `10 * labels[i].long() + Tensor(c).long()` is in `range(100)`, which is the number of neurons in the example.

Comment (Hananel-Hazan) [reply examples/cifar10/minimal_supervised_cifar10.py]: Ok, I see what you mean. You mixed up between time axis and neuron axis... its the opposite from what you wrote.
I'm pushing the updated code
